src/modules/product-category/entities/product-category.entity.ts

```diff
@@ -0,0 +1,34 @@
+import { AbstractBaseEntity } from "../../../entities/base.entity";
+import { Product } from "../../products/entities/product.entity";
+import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from "typeorm";
+
+// @Entity()
+// export class ProductCategory extends AbstractBaseEntity {
+// @Column({type:'text', unique: true})
+// name:string
+// @Column({type:'text', nullable: true})
+// description: string
+// @OneToMany(()=> Product, products => products.category)
+// product:Product
+// }
+
+@Entity('categories')
+export class ProductCategory extends AbstractBaseEntity{
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @Column({ length: 500 })
+  name: string;
+
+  @Column('text')
+  description: string;
+
+  @Column('text')
+  slug: string;
+
+  @Column({ type: 'uuid', nullable: true })
+  parent_id: string;
```

Comment:
Sorry, what does this column do? If its your primary key, then it should be removed. If its a foreign key, please define the relationship.

Reply:
I need to confirm from my other team member since i did not work on the product category directly

Reply:
just confirmed that it wont be needed for the stage of this application so i will be removing it.